driver-core/src/test/functional/com/mongodb/client/model/expressions/ComparisonExpressionsFunctionalTest.java

```diff
@@ -0,0 +1,186 @@
+    private final List<Expression> sampleValues = Arrays.asList(
+            ofRem(),
+            ofNull(),
+            of(0),
+            of(1),
+            of(""),
+            of("str"),
+            ofDocument(BsonDocument.parse("{}")),
+            ofDocument(BsonDocument.parse("{a: 1}")),
+            ofDocument(BsonDocument.parse("{a: 2}")),
+            ofDocument(BsonDocument.parse("{a: 2, b: 1}")),
+            ofDocument(BsonDocument.parse("{b: 1, a: 2}")),
+            ofDocument(BsonDocument.parse("{'':''}")),
+            ofIntegerArray(0),
+            ofIntegerArray(1),
+            ofBooleanArray(true),
+            of(false),
+            of(true),
+            of(new Date())
+    );
+
+    @Test
+    public void eqTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/eq/
+        assertExpression(
+                1 == 2,
+                of(1).eq(of(2)),
+                "{'$eq': [1, 2]}");
+        assertExpression(
+                false,
+                ofDocument(BsonDocument.parse("{}")).eq(ofIntegerArray()),
+                "{'$eq': [{'$literal': {}}, []]}");
+
+        // numbers are equal, even though of different types
+        assertExpression(
+                1 == 1.0,
+                of(1).eq(of(1.0)),
+                "{'$eq': [1, 1.0]}");
+        assertExpression(
+                1 == 1L,
+                of(1).eq(of(1L)),
+                "{'$eq': [1, { '$numberLong': '1' }]}");
+
+        // ensure that no two samples are equal to each other
+        for (int i = 0; i < sampleValues.size(); i++) {
+            for (int j = 0; j < sampleValues.size(); j++) {
+                if (i == j) {
+                    continue;
+                }
+                Expression u = sampleValues.get(i);
+                Expression v = sampleValues.get(j);
+                BsonValue evaluate = evaluate(u.eq(v));
+                if (evaluate.asBoolean().getValue()) {
+                    fail(i + " " + j + " should not equal");
+                }
+            }
+        }
+    }
+
+    @Test
+    public void neTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/ne/
+        assertExpression(
+                1 != 2,
+                of(1).ne(of(2)),
+                "{'$ne': [1, 2]}");
+    }
+
+    @Test
+    public void ltTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/lt/
+        assertExpression(
+                -1 < 1,
+                of(-1).lt(of(1)),
+                "{'$lt': [-1, 1]}");
+        assertExpression(
+                0 < 0,
+                of(0).lt(of(0)),
+                "{'$lt': [0, 0]}");
+
+        assertExpression(
+                true,
+                ofNull().lt(of(0)),
+                "{'$lt': [null, 0]}");
+
+        for (int i = 0; i < sampleValues.size() - 1; i++) {
+            int j = i + 1;
```

Comment:
We may also do `for (int j = i + 1; j < sampleValues.size(); j++)` to cover more pairs.